scripts/smoke/check.js

```diff
@@ -64,30 +76,61 @@ function checkExamples() {
 }
 
 /**
- * @param {string} examplePath
+ * 
+ * @param {string} tsconfigPath 
+ * @param {import('tsconfck').TSConfckParseOptions} options 
+ * @returns {Promise<import('tsconfck').TSConfckParseResult | 'invalid-config' | 'missing-config' | 'unknown-error'>}
  */
-function prepareExample(examplePath) {
-	const tsconfigPath = path.join('./examples/', examplePath, 'tsconfig.json');
-	const tsconfig = tsconfigResolverSync({ filePath: tsconfigPath, cache: false });
-	let originalConfig = undefined;
+async function safeParse(tsconfigPath, options = {}) {
+	try {
+		const parseResult = await parse(tsconfigPath, options);
 
-	if (tsconfig.exists) {
-		tsconfig.config.extends = 'astro/tsconfigs/strictest';
-		originalConfig = readFileSync(tsconfigPath).toString();
+		if (parseResult.tsconfig == null) {
+			return 'missing-config';
+		}
 
-		if (!tsconfig.config.compilerOptions) {
-			tsconfig.config.compilerOptions = {};
+		return parseResult;
+	} catch (e) {
+		// TODO: remove
+		console.error(e)
```

Comment:
Do we want to remove this? Personally I think it's also fine to leave it in for debugging, and if so we can remove the todo comment.